tools/fuse_installer.py

```
@@ -388,21 +390,48 @@ def install_package(package: ZipPackage, mods_dir: Path, replace: bool, dry_run:
+    # is never touched, and a fresh one is never left half-written for the game
+    # to load.
+    try:
+        extract_package(package, staging)
+    except BaseException:
+        shutil.rmtree(staging, ignore_errors=True)
+        raise
 
+    # Extraction succeeded. Back up any existing install, then swap staging into
+    # place with a rename on the same filesystem (atomic on Windows and POSIX).
+    if destination.exists():
         timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
```

Ruff (0.16.1)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)